images/docker-stacks-foundation/Dockerfile

```diff
@@ -117,7 +117,8 @@ RUN set -x && \
         'jupyter_core' && \
     rm -rf /tmp/bin/ && \
     # Pin major.minor version of python
-    mamba list python | grep '^python ' | tr -s ' ' | cut -d ' ' -f 1,2 >> "${CONDA_DIR}/conda-meta/pinned" && \
+    # https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-pkgs.html#preventing-packages-from-updating-pinning
+    mamba list --full-name 'python' | tail -1 | tr -s ' ' | cut -d ' ' -f 1,2 | sed 's/\.[^.]*$/.*/' >> "${CONDA_DIR}/conda-meta/pinned" && \
```

Comment:
I simplified a little bit the command here, `--full-name` is exactly what we need.
I used sed command from @ehfd issue: https://github.com/jupyter/docker-stacks/issues/2084#issue-2092258825, but removed extra space it was adding there.